src/autopas/containers/verletClusterLists/ClusterTowerBlock2D.h

```diff
@@ -349,24 +354,34 @@ class ClusterTowerBlock2D : public CellBorderAndFlagManager {
   const std::array<double, 2> &getTowerSideLengthReciprocal() const { return _towerSideLengthReciprocal; }
 
   bool cellCanContainHaloParticles(index_t index1d) const override {
-    // TODO
+    // Always true because towers cover the whole z dimension
     return true;
   }
 
   bool cellCanContainOwnedParticles(index_t index1d) const override {
-    // TODO
-    return true;
+    // check if the tower is strictly in the halo region
+    if (index1d < _firstOwnedTowerIndex or index1d > _lastOwnedTowerIndex) {
+      return false;
+    }
+    const auto index2D = towerIndex1DTo2D(index1d);
+    bool isHaloTower = false;
+    for (size_t i = 0; i < index2D.size(); ++i) {
```

Comment:
```suggestion
    for (size_t i = 0; i < 2; ++i) {
```
A bit more readable, and when 2D is literally in the name I don't feel like it adds any maintainability to use `.size()`

Reply:
The idea is that it is clear from the head of the loop what we iterate over. Also calling `size()` on an `std::array` should be "free" as it is a constant known at compile time. However, if you insist we can hard-code the `2`.

Reply:
I won't say that I insist, as my reasoning is only anecdotal: I had to second-guess what `index2D` was e.g. thinking that it might have been an array of indices - because `index2D.size()` is very clearly always going to be 2.

I would leave it up to you to decide if this anecdotal evidence is worthy of switching to hard-coding.

I do understand your point about making it clear what we are iterating over, and so if you would like to hard-code 2 I'd suggest adding a brief comment.